autotest/alg/warp.py

```diff
@@ -1608,38 +1590,36 @@ def test_warp_52():
 # Test Grey+Alpha
 
 
-def test_warp_53():
-
-    if sys.platform == 'darwin':
-        pytest.skip("Expected checksum should be updated for Mac")
-
-    for typestr in ('Byte', 'UInt16', 'Int16'):
-        src_ds = gdal.Translate('', '../gcore/data/byte.tif',
-                                options='-of MEM -b 1 -b 1 -ot ' + typestr)
-        src_ds.GetRasterBand(2).SetColorInterpretation(gdal.GCI_AlphaBand)
-        src_ds.GetRasterBand(2).Fill(255)
-        zero = struct.pack('B' * 1, 0)
-        src_ds.GetRasterBand(2).WriteRaster(10, 10, 1, 1, zero,
-                                            buf_type=gdal.GDT_Byte)
-        dst_ds = gdal.Translate('', src_ds,
-                                options='-of MEM -a_srs EPSG:32611')
-
-        for option in ('-wo USE_GENERAL_CASE=TRUE', ''):
-            # First checksum is proj 4.8, second proj 4.9.2
-            for alg_name, expected_cs in (('near', [3781, 3843]),
-                                          ('cubic', [3942, 4133]),
-                                          ('cubicspline', [3874, 4076]),
-                                          ('bilinear', [4019, 3991])):
-                dst_ds.GetRasterBand(1).Fill(0)
-                dst_ds.GetRasterBand(2).Fill(0)
-                gdal.Warp(dst_ds, src_ds,
-                          options='-r ' + alg_name + ' ' + option)
-                cs1 = dst_ds.GetRasterBand(1).Checksum()
-                cs2 = dst_ds.GetRasterBand(2).Checksum()
-                if cs1 not in expected_cs or cs2 not in [3903, 4138]:
-                    print(typestr)
-                    print(option)
-                    pytest.fail(alg_name)
+@pytest.mark.skipif(sys.platform == 'darwin',
+                    reason="Expected checksum should be updated for Mac")
```

Comment:
AFAIK the checksums should match across platforms? (I know this was there before)

Might be better as an xfail?

Reply:
It looks like only cubic fails:
```
alg/warp.py::test_warp_53[near-generalCase-Byte] XPASS (Expected che...) [ 91%]
alg/warp.py::test_warp_53[near-generalCase-UInt16] XPASS (Expected c...) [ 91%]
alg/warp.py::test_warp_53[near-generalCase-Int16] XPASS (Expected ch...) [ 91%]
alg/warp.py::test_warp_53[near-default-Byte] XPASS (Expected checksu...) [ 91%]
alg/warp.py::test_warp_53[near-default-UInt16] XPASS (Expected check...) [ 91%]
alg/warp.py::test_warp_53[near-default-Int16] XPASS (Expected checks...) [ 91%]
alg/warp.py::test_warp_53[cubic-generalCase-Byte] XFAIL (Expected ch...) [ 91%]
alg/warp.py::test_warp_53[cubic-generalCase-UInt16] XFAIL (Expected ...) [ 91%]
alg/warp.py::test_warp_53[cubic-generalCase-Int16] XFAIL (Expected c...) [ 91%]
alg/warp.py::test_warp_53[cubic-default-Byte] XFAIL (Expected checks...) [ 91%]
alg/warp.py::test_warp_53[cubic-default-UInt16] XFAIL (Expected chec...) [ 91%]
alg/warp.py::test_warp_53[cubic-default-Int16] XFAIL (Expected check...) [ 91%]
alg/warp.py::test_warp_53[cubicspline-generalCase-Byte] XPASS (Expec...) [ 91%]
alg/warp.py::test_warp_53[cubicspline-generalCase-UInt16] XPASS (Exp...) [ 91%]
alg/warp.py::test_warp_53[cubicspline-generalCase-Int16] XPASS (Expe...) [ 91%]
alg/warp.py::test_warp_53[cubicspline-default-Byte] XPASS (Expected ...) [ 91%]
alg/warp.py::test_warp_53[cubicspline-default-UInt16] XPASS (Expecte...) [ 91%]
alg/warp.py::test_warp_53[cubicspline-default-Int16] XPASS (Expected...) [ 91%]
alg/warp.py::test_warp_53[bilinear-generalCase-Byte] XPASS (Expected...) [ 92%]
alg/warp.py::test_warp_53[bilinear-generalCase-UInt16] XPASS (Expect...) [ 92%]
alg/warp.py::test_warp_53[bilinear-generalCase-Int16] XPASS (Expecte...) [ 92%]
alg/warp.py::test_warp_53[bilinear-default-Byte] XPASS (Expected che...) [ 92%]
alg/warp.py::test_warp_53[bilinear-default-UInt16] XPASS (Expected c...) [ 92%]
alg/warp.py::test_warp_53[bilinear-default-Int16] XPASS (Expected ch...) [ 92%]
```

Reply:
@rouault any recollections on why checksums might be different on macOS?

(was added in https://github.com/OSGeo/gdal/commit/6705917bb366522a4e77cca9c461badad3ed671f with several other macos checksum differences)

Reply:
Only thing that I spotted is that https://github.com/OSGeo/gdal/blob/0941b0f7aaa5bff5fd12a6fed03d4e385c4dcae4/gdal/alg/gdalchecksum.cpp#L75 uses `int` instead of `int32_t`/`GInt32` for the checksum value.

Reply:
> @rouault any recollections on why checksums might be different on macOS?

I'm not sure why. Might be related to using a different PROJ version and/our missing datum shift grids and/or subtle numeric differences somewhere...